lib/proxy/pluggable.rb

```diff
@@ -1,131 +1,137 @@
 module Proxy::Pluggable
-  def self.included(base)
-    base.send(:include, InstanceMethods)
-    base.send(:extend, ClassMethods)
+  attr_reader :plugin_name, :version, :after_activation_blk, :class_loader
+
+  # Methods below define DSL for defining plugins
+  def after_activation(&blk)
+    @after_activation_blk = blk
   end
 
-  module InstanceMethods
-    def plugin_name
-      self.class.plugin_name
-    end
+  def requires(plugin_name, version_spec)
+    self.dependencies << ::Proxy::Dependency.new(plugin_name, cleanup_version(version_spec))
+  end
 
-    def version
-      self.class.version
-    end
+  def bundler_group(name)
```

Comment:
Seems to be fixed at https://github.com/theforeman/smart-proxy/pull/390/files#diff-192cbec985fe34a2e50b9569df0975f4R43